src/utils.ts

```diff
@@ -452,13 +467,130 @@ export class Utils {
         }
         return;
     }
+
+    /**
+     * Generates GitHub workflow Summary
+     * This function runs as part of post-workflow cleanup function,
+     * collects existing section markdown files generated by the CLI,
+     * and constructs a single markdown file, to be displayed in the GitHub UI.
+     */
+    public static async generateWorkflowSummaryMarkdown() {
+        // This is the file path that the markdown should be copied to.
+        // In order for GitHub to display this as a job summary in the UI.
+        const githubStepSummaryFilePath: string | undefined = process.env.GITHUB_STEP_SUMMARY;
+        if (!githubStepSummaryFilePath) {
+            core.debug('GITHUB_STEP_SUMMARY is not set. Workflow summary will not be generated.');
+            return;
+        }
+
+        try {
+            // Read all sections and construct the final markdown file
+            const markdownContent: string = await this.readCLIMarkdownSections();
+            if (markdownContent.length == 0) {
+                core.debug('No job summaries sections found. Workflow summary will not be generated.');
+                return;
+            }
+            await fs.writeFile(githubStepSummaryFilePath, markdownContent);
+            // Clear files
+            await this.clearJobSummaryDir();
+        } catch (error) {
+            core.warning(`Failed to generate Workflow summary: ${error}`);
+        }
+    }
+
+    // Each section should prepare a file called markdown.md. 
+    // This function reads each section file and wraps it with a markdown header
+    private static async readCLIMarkdownSections(): Promise<string> {
+        const outputDir: string = Utils.getJobOutputDirectoryPath();
+        let fileContent: string = '';
+
+        for (const sectionName of Utils.JOB_SUMMARY_MARKDOWN_SECTIONS_NAMES) {
+            fileContent += await Utils.readSummarySection(outputDir, sectionName);
+        }
+
+        // No section was added, return empty string
+        if (fileContent == '') {
+            return '';
+        }
+        // Add the main header
+        return this.getMarkdownHeader() + fileContent;
+    }
+
+    private static async readSummarySection(outputDir: string, section: string) {
+        let content :string = '';
+        try {
+            const fullPath: string = path.join(outputDir, section, 'markdown.md');
+            content = await fs.readFile(fullPath, 'utf-8');
+        } catch (error) {
+            core.debug(`Section ${section} not found or empty, skipping...`);
```

Comment:
If the file doesn't exists, it's okay. Because not every run all the section are guaranteed to be created.
Maybe i'll make it more clear that this is the intention 👍